protected/humhub/components/Module.php

```diff
@@ -239,21 +243,28 @@ public function getIsActivated(): bool
     /**
      * Enables this module
      *
-     * @return boolean
+     * @return bool|null Result of migration or null if beforeEnable() returned false (since v1.16)
+     * @throws InvalidConfigException
      */
     public function enable()
     {
         Yii::$app->moduleManager->enable($this);
-        $this->migrate();
 
-        return true;
+        $result = $this->getMigrationService()->migrateUp();
+
+        if ($result === false) {
+            Yii::$app->moduleManager->disable($this);
```

Comment:
Since `disable` deletes data, I would like to drop this. Until we have a suspend or similar.

Not sure: But maybe in some reinstall situations the enable is executed with existing data.

Reply:
No, as explained [before](https://github.com/humhub/humhub/pull/6550#discussion_r1419434181), the `$moduleManager->disable` does _not_ delete any data, it only removes the module from the the `module_enable` table and the `Yii::$app`'s modules again. It's the `Module::disable()` that deletes data.

Why should the module be enabled in the `module_enable` table, if the migrations fail?

Maybe we should only "enable" the module in the module manager only _after_ the migration and only if they are successful?

Reply:
Ok, no worries.

What do you think of the following?

> Maybe we should only "enable" the module in the module manager only after the migration and only if they are successful?

Reply:
@martin-rueegg Yeah, I really like this idea. Maybe with error message: 

```php

    public function enable()
    {
        $migrationService = $this->getMigrationService();

        if (!$migrationService->migrateUp()) {
            Yii::error('Could not enable module. Database Migration failed! Result: ' .
                $migrationService->getLastMigrationOutput(), $this->id);
            return false;
        }

        Yii::$app->moduleManager->enable($this);

        return true;
    }

```

Reply:
Sure, I can add an error message.

The migration's output is sent to the log in any way. As warning, if it's successful. As error if it failed:
https://github.com/humhub/humhub/pull/6550/files#diff-381c812c100b93c52f71ee2cd25c98821a7436b5d1223f2f2d96059fcc05950dR237-R241

But it may be worth adding an other error.

Reply:
Ok, then the migration output does not have to be included in the error message again. But a hint as to why the module was not enabled would be good.

Reply:
Already there:
https://github.com/humhub/humhub/blob/374501b9d30fe5c5df634870a2c7e690e9a41f16/protected/humhub/components/Module.php#L251-L259

Reply:
Done: [Add hasMigrations(Pending)()](https://github.com/humhub/humhub/pull/6550/commits/bfe92a47fa159d5cc7d95c707074217085ba4ba1);